### PR TITLE
doc/rbd/rbd-exclusive-locks: mention incompatibility with advisory locks

### PR DESCRIPTION
**Pull request by idryomov**
This is due to both using the same name for the lock at the RADOS level ("rbd_lock").

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by ajarr — APPROVED** [review]

**Comment by idryomov** [comment]
Spotted a missing backtick, now fixed.

File: doc/rbd/rbd-exclusive-locks.rst
Comment (zdover23) [inline doc/rbd/rbd-exclusive-locks.rst:65]: s/``rbd lock add``/the ``rbd lock add``/

This is one of these tricky situations where the article is used in English, but the only way to explain why is to stand upon fingerspitzengefühl.

Comment (zdover23) [reply doc/rbd/rbd-exclusive-locks.rst:65]: Anyway, this isn't a showstopper. I'm approving it. Does this get backported to S, R, and Q?

**Review by zdover23 — APPROVED** [review]

File: doc/rbd/rbd-exclusive-locks.rst
Comment (idryomov) [reply doc/rbd/rbd-exclusive-locks.rst]: > s/`rbd lock add`/the `rbd lock add`/

Done.

> Does this get backported to S, R, and Q?

Yes, to all supported releases.

**Comment by ajarr** [comment]
Please don't merge. I've another question

**Comment by ajarr** [comment]
In doc/man/8/rbd.rst, I see "exclusive" being used when referring to advisory locks. Here, https://github.com/ceph/ceph/blob/main/doc/man/8/rbd.rst?plain=1#L444 and  https://github.com/ceph/ceph/blob/main/doc/man/8/rbd.rst?plain=1#L999 .  Do we want to mention that these are advisory locks somewhere in that man page?

**Comment by ajarr** [comment]
Answer makes sense to me. I think this PR is ready to merge once the doc build passes.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/58863 - Squid backport
https://github.com/ceph/ceph/pull/58864 - Reef backport
https://github.com/ceph/ceph/pull/58865 - Quincy backport

**Pull request merged**
